### PR TITLE
FossId: Explain why the class implements `PackageScannerWrapper`

### PR DESCRIPTION
**Pull request by mnonnenmacher**

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/fossid/FossId.kt [hunk 1]
```diff
@@ -86,6 +87,10 @@ import org.ossreviewtoolkit.utils.ort.showStackTrace
  *
  * This scanner can be configured in [ScannerConfiguration.options]. For the options available and their documentation
  * refer to [FossIdConfig].
+ *
+ * This scanner was implemented before the introduction of [provenance based scanning][ProvenanceScannerWrapper].
+ * Therefore it implements the [PackageScannerWrapper] interface for backward compatibility, even though FossID itself
+ * gets a Git repository as input and would be a good match for [ProvenanceScannerWrapper].
```
Comment (nnobelis) [inline scanner/src/main/kotlin/scanners/fossid/FossId.kt:93]: Not sure about the wording here.
FossID takes a Git repository URL as input and clones the repository.
Does it still qualifies as [ProvenanceScannerWrapper] ?

Comment (mnonnenmacher) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:93]: Yes, the input for `ProvenanceScannerWrapper` is a source code location, while the input for `PackageScannerWrapper` is a package.

Comment (nnobelis) [reply scanner/src/main/kotlin/scanners/fossid/FossId.kt:93]: Then maybe reword with `...gets a Git repository URL ad input...`

**Review by fviernau — APPROVED** [review]

**Review by sschuberth — APPROVED** [review]
